src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java

```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
         CaseJudgementUpdate.updateCaseWithJudgement(caseData, judgementType);
     }
 
-    private void updateRespondentRep(CaseData caseData, RepresentedTypeR representedType) {
+    private void updateRespondentRep(CaseData caseData, UpdateDataModel updateDataModel) {
 
-        //case 1: update is adding
+        if (caseData.getRespondentCollection() != null) {
 
-        //case 2: update is removing
+            RepresentedTypeR representedType = updateDataModel.getRepresentedType();
+            String isRespondentRepRemovalUpdate = updateDataModel.getIsRespondentRepRemovalUpdate();
+            boolean respondentRepRemovalUpdate = isRespondentRepRemovalUpdate.equals(YES);
 
-        if (caseData.getRespondentCollection() != null) {
             Optional<RespondentSumTypeItem> respondentSumTypeItemOptional =
                     caseData.getRespondentCollection().stream()
                             .filter(respondentSumTypeItem ->
                                     respondentSumTypeItem.getValue().getRespondentName()
                                             .equals(representedType.getRespRepName()))
                             .findAny();
 
-            if (respondentSumTypeItemOptional.isPresent()) {
+            if(respondentSumTypeItemOptional.isPresent()){
+                if (!respondentRepRemovalUpdate) {
+                    addRespondentRepUpdates(caseData, updateDataModel);
+                }else {
+                    addRespondentRepRemovalUpdate(caseData, representedType);
+                }
+            }
+        }
+    }
 
-                if (caseData.getRepCollection() != null) {
+    private void addRespondentRepRemovalUpdate(CaseData caseData, RepresentedTypeR representedType){
+        if(representedTypeRItemExists(caseData, representedType)){
+            getExistingRepresentedTypeRItem(caseData, representedType).setValue(new RepresentedTypeR());
+        }
+    }
 
-                    boolean found = false;
+    private void addRespondentRepUpdates(CaseData caseData, UpdateDataModel updateDataModel){
+        boolean caseDataHasRep = (caseData.getRepCollection() != null);
+        RepresentedTypeR representedType = updateDataModel.getRepresentedType();
```

Comment:
representedType is only used in the following if statement so it should be declared in there too